how-to-guides/celestia-app-security.md

```diff
@@ -0,0 +1,282 @@
+ipv4 blacklist full ipset:blocked_ips ipset:blocked_nets
+
+# Define your external network interface
+interface4 enp5s0 internet
+    protection strong
+
+    # Allow SSH traffic on port 22 (TCP) for public access
+    server ssh accept
+
+    # Allow P2P traffic on port 26656 (TCP) with DDoS protection
+    server custom p2p tcp/26656 default accept
+    protection syn-floods 20/sec 50  # Protect against SYN flood attacks
+
+    # Allow RPC traffic on port 26657-26659 (TCP) with DDoS protection
+    server custom rpc tcp/26657:26659 default accept
+    protection syn-floods 20/sec 50  # Protect against SYN flood attacks
+
+    # Allow Prometheus metrics on port 26660 (limit to trusted IPs or internal network) with DDoS protection
+    server custom prometheus tcp/26660 default accept src 192.168.1.0/24
+    protection syn-floods 10/sec 20  # Protect against SYN flood attacks
+
+    # Allow gRPC communication on port 9090 (TCP) with DDoS protection
+    server custom grpc tcp/9090 default accept
+    protection syn-floods 15/sec 30  # Protect against SYN flood attacks
+
+    # Allow gRPC communication on port 9091 (TCP) with DDoS protection
+    server custom grpc2 tcp/9091 default accept
+    protection syn-floods 15/sec 30  # Protect against SYN flood attacks
+
+    # Allow REST API access on port 1317 (TCP) with DDoS protection
+    server custom rest tcp/1317 default accept
+    protection syn-floods 20/sec 40  # Protect against SYN flood attacks
+
+    # Additional Ports
+    server custom 11656 tcp/11656 default accept
+    server custom 11656 udp/11656 default accept
+    server custom 11065 tcp/11065 default accept
+    server custom 11065 udp/11065 default accept
+    server custom 36656 tcp/36656 default accept
+    server custom 36656 udp/36656 default accept
+    server custom 6065 tcp/6065 default accept
+    server custom 6065 udp/6065 default accept
+    server custom 9099 tcp/9099 default accept
+    server custom 9099 udp/9099 default accept
+
+    # Allow traffic on port 2121 for both TCP and UDP (separate rules for TCP and UDP)
+    server custom 2121 tcp/2121 default accept
+    protection syn-floods 20/sec 40  # Protect against SYN flood attacks
+    server custom 2121 udp/2121 default accept
+    protection syn-floods 20/sec 40  # Protect against SYN flood attacks
+
+    # Allow all traffic from localhost without specifying ports
+    server all accept src 127.0.0.1
+
+    # Allow all traffic from internal network without specifying ports
+    server all accept src 192.168.1.0/24
+
+    # Allow outgoing traffic for general internet access
+    client all accept
+
+# Block all IPv6 traffic
+interface4 enp5s0 ipv6
+    server all drop
+```
+
+::: tip
+
+Currently bind to eth0, make sure you bind it to the correct interface or bind all
+
+```interface any world```
+
+:::
+
+### Key Configuration Details
+
+- P2P (26656): Allows inbound P2P traffic, essential for Celestia node synchronization.
+
```

Comment:
no blanks needed between items in this list

Reply:
Resolved